### DISCUSSION TITLE
What is marking a field as dirty mean in pimcore?

### DISCUSSION THREAD

**Discussion by prunepal333** [pimcore/pimcore#14385] [General]
Can someone elaborate with examples about what does dirty field mean in case of pimcore?

**Comment by SamyMP** [comment]
I think dirty means the object version of the dataobject marked as dirty is not the same as the published version. It implies that at least one field of the dataobject has been changed

**Reply by NiklasBr** [reply]
It'd done on the [field level](https://github.com/pimcore/pimcore/blob/11.x/models/Element/DirtyIndicatorInterface.php), but only on a few fields, so it is unclear why only those fields?

```php
        if (!$isEqual) {
		$this->markFieldDirty("TheFieldName", true);
	}
```
